src/app/organization/offices/office-tree.service.ts

```diff
@@ -0,0 +1,80 @@
+/** Angular Imports */
+import { Injectable } from '@angular/core';
+
+/** rxjs Imports */
+import { BehaviorSubject } from 'rxjs';
+
+/** Custom Components */
+import { OfficeNode } from './office-node.model';
+
+/**
+ * Office tree service.
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class OfficeTreeService {
+
+  /** Office data. */
+  officesData: any;
+  /** Offices tree data behavior subject to represent offices tree nodes. */
+  treeDataChange = new BehaviorSubject<OfficeNode[]>([]);
+
+  /**
+   * Gets the offices tree nodes.
+   */
+  get treeData(): OfficeNode[] { return this.treeDataChange.value; }
+
+  constructor() {  }
+
+  /**
+   * Builds the offices tree and emits the value.
+   * @param {any} officesData Offices data.
+   */
+  initialize(officesData: any) {
+    const treeData = this.buildOfficeTree(officesData);
+    this.treeDataChange.next(treeData);
+  }
+
+  /**
+   * Builds and returns the offices tree.
+   * @param {any} officeData Offices data.
+   * @returns {OfficeNode} Offices tree nodes.
+   */
+  buildOfficeTree(officesData: any): OfficeNode[] {
+    const officeTree: OfficeNode[] = [];
+
+    // Header nodes
+    officeTree.push(new OfficeNode(officesData[0].name, officesData[0].id));
```

Comment:
I know that I have suggested you to do this myself, but lets make sure that the thing that we push first is the Main Head Office, incase the array comes sorted so that the Main Head Office is not the first element.
```suggestion
    const mainOffice = officesData.find((office: any) => office.hierarchy === '.');
    officeTree.push(new OfficeNode(mainOffice.name, mainOffice.id));
```

Reply:
I am sorry if this does not run, I am quite busy at the moment so I couldn't personally test this myself.

Reply:
It is working, no errors present. Thanks for the suggestion :)